design/data-mover.md

```diff
@@ -0,0 +1,73 @@
+# Velero Data Mover
+
+## Abstract
+Velero currently does not handle data movement of snapshotted data.  We would like to have the ability to move data that has been snapshotted in a storage system or other [Astrolabe Protected Entity](https://github.com/vmware-tanzu/astrolabe) provider into the backup storage location specified.
+
+## Background
+Velero has two options for backing up persistent volumes:
+Using snapshotting APIs provided by the storage provider (for example, using the VolumeSnapshotter)
+Filesystem level backup using restic
+
+Snapshotting capabilities provided by storage providers generally provide crash consistency of volumes, however they are not portable and are typically locked to the particular cloud provider and region in which the snapshot was taken.
+Although restic provides portability of data, it is a filesystem level backup and is not crash-consistent by default.
+It requires the quiescing of data prior to backup which impacts the availability of an application.
+
+Velero also has beta support for taking snapshots using CSI.
+Without data movement though, there is no guarantee that the snapshots are durable as it is dependent on the storage provider.
+
+Extraction of data from snapshots is something that is already available in the [Velero Plugin for vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere).
+This plugin makes use of the Astrolabe APIs to implement a vSphere FCD Protected Entity.
+This Protected Entity interface provides data readers and writers for the disks.
+Once a snapshot is created, the data reader is used to read the data from the disk and it is written to an S3 compatible object store.
+
+The Astrolabe API can be implemented for resources other than volumes, for example, databases.
```

Comment:
Is this relevant to this design?  Will data mover only cover the data in `snapshot` or will we consider the dump file also a `snapshot`?

Currently, in velero the snapshot is the narrower concept.